### PR TITLE
feat(Homebrew-Shapeshift): add new subviews to exchange create view

### PR DESCRIPTION
**Pull request by woudini**
## Objective
Add additional subviews to Homebrew Shapeshift view as specified by design. 

## Description

Added a new class `AssetConversionRateView` and its viewModel `AssetConversionRateViewModel`, and used it conditionally in `ExchangeCreateView` with the parameter `withConversionView`.

## How to Test

To test Homebrew, cherry-pick 8cfb01692a4e2043ad11a1d8622813e0e3f5001f and go to Exchange
To test Shapeshift, cherry-pick 969038f31bd4451ff004ff511616d0940394f004 on top of 8cfb01692a4e2043ad11a1d8622813e0e3f5001f and go to Exchange

The above commits enable access to Shapeshift and add a mock view model for the `AssetConversionRateView`.

## Screenshot

simulator screen shot - iphone 6 - 2018-08-21 at 23 01 21

## Related Information

* Ticket Number: IOS-1159

## Merge Checklist

- [x] The PR uses a title supported by [.changelogrc](https://github.com/blockchain/My-Wallet-V3-iOS/blob/dev/.changelogrc#L6...L69).
- [x] Areas of technical debt are marked with a `// TICKET:` comment that includes a ticket number.
- [x] All unit tests pass.
- [ ] You have added unit tests.
- [x] New files added are within the correct directory. (e.g. if a file is required for unit tests to compile, be sure it is added to the tests target.)

### CODE REVIEW

File: Blockchain/Extensions/UIView+Center.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//  UIView+Center.swift
+//  Blockchain
+//
+//  Created by kevinwu on 8/18/18.
+//  Copyright © 2018 Blockchain Luxembourg S.A. All rights reserved.
+//
+
+import Foundation
+
+public extension UIView {
```
Comment (bchrisarriola) [inline Blockchain/Extensions/UIView+Center.swift]: Nice extensions 👍

File: Blockchain/Homebrew/Exchange/HomebrewExchangeCreateViewController.swift [hunk 2]
```diff
@@ -25,13 +25,15 @@ class HomebrewExchangeCreateViewController: UIViewController {
...
-        let exchangeCreateView = ExchangeCreateView(frame: view.bounds)
+        exchangeCreateView = ExchangeCreateView(frame: view.bounds)
         view.addSubview(exchangeCreateView)
 
         exchangeCreateView.setup(
+            withConversionView: true,
             delegate: self,
             navigationController: self.navigationController as! BCNavigationController
         )
+        LoadingViewPresenter.shared.hideBusyView()
```
Comment (bchrisarriola) [inline Blockchain/Homebrew/Exchange/HomebrewExchangeCreateViewController.swift]: Why is this here? This should be placed elsewhere (i.e. whichever component called `LoadingViewPresenter.shared.showBusyView()`)

File: Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+//  AssetConversionRateView.swift
+//  Blockchain
+//
+//  Created by kevinwu on 8/18/18.
+//  Copyright © 2018 Blockchain Luxembourg S.A. All rights reserved.
+//
+
+import Foundation
+
+class AssetConversionRateView: UIView {
```
Comment (bchrisarriola) [inline Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift]: Why is this created programmatically? This should be moved to be created using a xib. We're trying to move views away from programmatically creating them for maintainability/scalability reasons unless there's no way around it or it is very expensive/involved to do so.

Comment (bchrisarriola) [reply Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift]: Taking inspiration from `Storyboardable`, you can:
1. Create a protocol `Nibable` in a file called `Nibable.swift`
2. Create a .xib titled `AssetConversionRateView`
3. Instantiate an `AssetConversionRateView` using `makeFromNib()`

Code for Nibable:

```
protocol Nibable: class {
    static var defaultNibName: String { get }
}

extension Nibable where Self: UIView {
    static var defaultNibName: String {
        return String(describing: self)
    }

    static func makeFromNib() -> Self {
        let nib = UINib(nibName: defaultNibName, bundle: Bundle.main)
        let contents = nib.instantiate(withOwner: nil, options: nil)
        return contents.first { $0 is Self } as! Self
    }
}

extension UIView: Nibable { }```

Comment (thisisalexmcgregor) [reply Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift]: Actually, you can also use `NibBasedView`. 

Just subclass that, and make a xib with the same name. In the `File's Owner` set it as whatever class you're making. In this case it'd be `AssetConversionRateView`. I did this with `PrimaryButtonContainer` if you need an example.

File: Blockchain/Extensions/UIView+Center.swift (hunk 1 above)
Comment (thisisalexmcgregor) [inline Blockchain/Extensions/UIView+Center.swift]: Autolayout? 

Actually if you use `NibBasedView` you can probably do a lot of this in a xib vs. having to do all the below programmatically.

File: Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift (hunk 3 above)
Comment (woudini) [reply Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift]: I don't have a good reason, just wanted to get it done quickly. Thanks for the code snippet, it was a lot easier than I thought to add. @thisisalexmcgregor I couldn't find a manual instantiation of `PrimaryButtonContainer` which is what I need in order to use it in `ExchangeCreateView`.

Comment (bchrisarriola) [reply Blockchain/Homebrew/Exchange/Views/AssetConversionRateView.swift]: Forgot about `NibBasedView`! I do like `Nibable` though since it's a protocol and won't need to rely on inheritance (`NibBasedView` can also conform to `Nibable` for consolidation)

File: Blockchain/Homebrew/Exchange/HomebrewExchangeCreateViewController.swift (hunk 2 above)
Comment (woudini) [reply Blockchain/Homebrew/Exchange/HomebrewExchangeCreateViewController.swift]: This should have been in the test branch - will remove.

File: Blockchain/Extensions/UIView+Center.swift (hunk 1 above)
Comment (woudini) [reply Blockchain/Extensions/UIView+Center.swift]: Removed extension
